### PR TITLE
feat: unlink chapter calendar

### PR DESCRIPTION
**Pull request by gikf**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Ref #1896
Ref #2168

---

- Adds button for unlinking calendar from chapter dashboard.
  - Calendar id is removed from chapter.
  - Calendar event ids are removed from events in the chapter.
- Currently button is displayed always when calendar is linked. Related issue suggests displaying it under some conditions. I'm not sure if that's needed, if we know that access to calendar is lost, need to confirm it explicitly would be unnecessary.
- What bothers me a bit is refetching all `DASHBOARD_EVENTS`, because of unlinking events.

### CODE REVIEW

File: client/src/modules/dashboard/Chapters/pages/ChapterPage.tsx [hunk 1]
```diff
@@ -133,7 +188,19 @@ export const ChapterPage: NextPageWithLayout = () => {
               {loadingCalendar ? (
                 <Spinner size="sm" />
               ) : data.dashboardChapter.calendar_id ? (
-                <CheckIcon boxSize="5" />
+                <>
+                  <CheckIcon boxSize="5" />
+                  {checkPermission(user, Permission.ChapterEdit, {
```
Comment (ojeytonwilliams) [inline client/src/modules/dashboard/Chapters/pages/ChapterPage.tsx:193]:
```suggestion
                  {checkPermission(user, Permission.ChapterCreate, {
```
This is fairly serious and (hopefully) not something people will often need, so I think it should be restricted to instance owners.

File: server/src/controllers/Chapter/resolver.ts [hunk 2]
```diff
@@ -201,6 +201,22 @@ export class ChapterResolver {
     }
   }
 
+  @Authorized(Permission.ChapterEdit)
```
Comment (ojeytonwilliams) [inline server/src/controllers/Chapter/resolver.ts:204]:
```suggestion
  @Authorized(Permission.ChapterCreate)
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Since accidentally triggering this would be incredibly annoying, I think we need some fairly heavy guardrails.  For instance, if chapter still has access to the calendar, it shouldn't be possible to unlink it.

So, it should try to get the calendar first and, if it can, just tell the user that nothing needs to be done.

**Comment by gikf** [comment]
Hm, so maybe making explicit test first, before giving ability to unlink, would be better after all. It would be strange showing the _Unlink_ button, asking if unlink it, and then go: _yeah, no_

**Comment by ojeytonwilliams** [comment]
I know what you mean, but it doesn't seem right to spam the Google api just because a user may need to see a button.

How about handling that with explanatory text?

> If the Google Calendar for this chapter no-longer exists, you can use the Unlink button to stop Chapter from trying to connect to it. Note: this will do nothing if the Calendar exists and Chapter has access.

It's a bit wordy, but this is an edge case that (hopefully) won't come up much.

**Comment by ojeytonwilliams** [comment]
Ah, that makes sense. Thanks for clarifying!

> Test -> Ask to confirm test, with some explanation -> If test fails display Unlink button, or directly use new modal to ask about unlinking.

That sounds perfect.

**Review by ojeytonwilliams — APPROVED** [review]
Cheers, @gikf 

LGTM :+1:
